clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -4381,6 +4411,34 @@ public boolean hasAllPartitionReportedCompleted() {
     return true;
   }
 
+  public boolean isSeparatedRealtimeTopicEnabled() {
+    return isSeparatedRealtimeTopicEnabled;
+  }
+
+  PubSubTopicPartition resolveTopicPartitionWithKafkaURL(
+      PubSubTopic topic,
+      PartitionConsumptionState partitionConsumptionState,
+      String kafkaURL) {
+    PubSubTopicPartition pubSubTopicPartition = partitionConsumptionState.getSourceTopicPartition(topic);
+    if (topic.isRealTime() && getKafkaClusterUrlResolver() != null
+        && !kafkaURL.equals(getKafkaClusterUrlResolver().apply(kafkaURL))) {
+      PubSubTopic separateRealTimeTopic =
+          getPubSubTopicRepository().getTopic(Version.composeSeparateRealTimeTopic(topic.getStoreName()));
+      pubSubTopicPartition =
+          new PubSubTopicPartitionImpl(separateRealTimeTopic, partitionConsumptionState.getPartition());
+    }
+    LOGGER.info("Resolved topic-partition: {} from kafkaURL: {}", pubSubTopicPartition, kafkaURL);
+    return pubSubTopicPartition;
+  }
+
+  PubSubTopicRepository getPubSubTopicRepository() {
```

Comment:
Seems this test-only PubSubTopicRepository is not necessary, PubSubTopicImpl could handle equal check by comparing String name

Reply:
Hmm, actually this is used in main code as well. PubSubTopicImpl constructor is package private by design so it is not accessible from SIT class.